app/controllers/api_secrets_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ApiSecretsController < ApplicationController
+  before_action :set_user
+
+  def create
+    @secret = ApiSecret.new(description: params[:description], user_id: @user.id)
+    if @secret.save
+      flash[:notice] = "Your access token has been generated: #{@secret.secret}. Be sure to copy it to somewhere safe now. You won’t be able to see it again!"
+      redirect_back(fallback_location: root_path)
+    end
+  end
+
+  def destroy
+    @secret = ApiSecret.find_by_id(params[:id])
```

Comment:
Note to self that this is missing a check for whether the user actually owns this secret to prevent users from deleting other users' secrets

Reply:
You should use [pundit](https://github.com/varvet/pundit) to handle authorization in the controller.  Check out some other controllers for examples. If you haven't used pundit before it's a bit unintuitive at first.